### ISSUE TITLE
Add to "How to Contribute" checking git artifact that PDFs knit (and anything else someone should know/do when contributing)

### ISSUE THREAD

**Issue by lemireg** [FredHutch/VISCtemplates#227]
![image](https://github.com/user-attachments/assets/c2e9d689-095d-4a61-8020-7dc1a72a9739)

^^ is there a benefit to checking git artifacts to see things knit (binary yes/no) (all that is checked in tests right?), or is the benefit of git artifacts only in seeing the content of the PDF.

This may just be an extra line in the "How to Contribute" page: https://github.com/FredHutch/VISCtemplates/blob/main/CONTRIBUTING.md

**Comment by slager** [comment]
Yes, the unit tests should fail if a report doesn't knit. The main intent of the artifacts is to be able to visually inspect the knit documents, and to be able to look at the intermediate/log/test files when debugging.

**Comment by lemireg** [comment]
> I started a new branch for updating CONTRIBUTING.md: https://github.com/FredHutch/VISCtemplates/blob/update-how-to-contribute/CONTRIBUTING.md
> 
> I just put a description at a very high level of the CI. [@lemireg](https://github.com/lemireg) and [@slager](https://github.com/slager) do you think we should add more details?

Comments/suggestions:
- I like the modification in organization: ordered bugs first and improvements second.
- I like the additional description of MRE. I'm wondering if the title "Guidelines for filing GitHub issues" should be "Guidelines for creating MRE"?
- "6 Create a PR" could mention the option to create a "draft PR" if it's still in development
- Question about "6. Please follow the official `tidyverse` style." <-- in package code I've noticed we don't really do this? so I'm wondering about mentioning it here (seems like people use base R as much as possible which I interpreted to be because we want to avoid external dependencies when possible)
- I like the addition of the CI description and picture! After talking to Dave about these runs I understand them better. With that in mind, to help someone debug this sentence about CI could be added: "The CIs are running tests on varying combinations of operating system (OS), R studio version, and R-version. For example `R-Cmd-check/ubuntu-latest (4.0.4)` runs the current code on linux operating system on the latest R studio version with R-version 4.0.4" <-- please correct me if I'm mixing this description up.

This looks great @kelliemac thank you for the improvements!

**Comment by kelliemac** [comment]
thanks @lemireg! I incorporated a couple of your suggestions right away, the remaining ones are now:

- if the title "Guidelines for filing GitHub issues" should be "Guidelines for creating MRE"?
- about "6. Please follow the official tidyverse style." <-- in package code I've noticed we don't really do this? so I'm wondering about mentioning it here (seems like people use base R as much as possible which I interpreted to be because we want to avoid external dependencies when possible)

curious to hear thoughts from @mayerbry about these and any other aspects

**Comment by slager** [comment]
One quick note is that I think in this context "tidyverse style" `!=` using the tidyverse instead of base R, and likely refers to the [tidyverse style guidelines](https://style.tidyverse.org/).

**Comment by kelliemac** [comment]
thanks @slager, I agree. I just added a note about this distinction to CONTRIBUTING.md in the new branch, to make this clearer.

**Comment by kelliemac** [comment]
I'm going to open a PR now for that branch, so that we can move some of this discussion over there.
